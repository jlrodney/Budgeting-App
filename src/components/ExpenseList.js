import React from 'react';
import { connect } from 'react-redux';
import ExpenseItem from './ExpenseItem';

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
          props.expenses.map((expense) => {
            return <ExpenseItem key={expense.id} {...expense} />;
          })
        )
    }
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses
  };
};

export default connect(mapStateToProps)(ExpenseList);
