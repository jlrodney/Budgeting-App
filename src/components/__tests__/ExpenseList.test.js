import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../ExpenseList';
import moment from 'moment';

const expenses = [{
  id: '1',
  description: 'Elictricity',
  note: '',
  amount: 180,
  createdAt: 0
}, {
  id: '2',
  description: 'Lunch',
  note: '',
  amount: 5,
  createdAt: moment(0).subtract(4, 'days').valueOf()
}, {
  id: '3',
  description: 'Credit Card debt interest',
  note: '',
  amount: 60,
  createdAt: moment(0).add(4, 'days').valueOf()
}];
describe('Expense list', () => {
  it('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ExpenseList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
