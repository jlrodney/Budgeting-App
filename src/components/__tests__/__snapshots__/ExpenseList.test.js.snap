// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense list should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`Expense list should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseItem
    amount={180}
    createdAt={0}
    description="Elictricity"
    id="1"
    key="1"
    note=""
  />
  <ExpenseItem
    amount={5}
    createdAt={-345600000}
    description="Lunch"
    id="2"
    key="2"
    note=""
  />
  <ExpenseItem
    amount={60}
    createdAt={345600000}
    description="Credit Card debt interest"
    id="3"
    key="3"
    note=""
  />
</div>
`;
